iot_api/user_api/repository/Tags.py

```diff
@@ -0,0 +1,117 @@
+import json
+from flask import request, abort, jsonify
+from flask_jwt_extended import get_jwt_identity
+from flask_restful import Resource
+from flask_jwt_extended import jwt_required, get_jwt_identity
+
+import iot_logging
+log = iot_logging.getLogger(__name__)
+
+from sqlalchemy import func, or_, distinct
+from sqlalchemy.sql import select, expression, text
+
+from iot_api.user_api import db
+from iot_api.user_api.models import Tag, DeviceTag, GatewayTag
+from iot_api.user_api.model import Device, Gateway
+from iot_api.user_api.repository import Devices, Gateways
+
+
+def list_all(organization_id):
+    """
+    List all tags of an organization.
+    """
+    result = db.session.query(Tag).filter(Tag.organization_id==organization_id).all()
+    return result if result else []
+
+def create(name, color, organization_id):
+    """
+    Create a new tag with the given name, color and organization_id.
+    """
+    tag = Tag(name=name, color=color, organization_id=organization_id)
+    db.session.add(tag)
+    db.session.commit()
+
+def get_with(tag_id, organization_id):
```

Comment:
Because if we cannot retrieve the tag we cannot update/delete/etc. This exception is handled in the endpoint, an error message is logged (for debugging) and a generic 400 is responded to the front.